index.js

```diff
@@ -1,36 +1,45 @@
 const entries = require('object.entries')
 
-function targetFor (source, destructive) {
+function defaultOnArray () { return [] }
+function defaultOnObject () { return {} }
+
+function targetFor (source, key, fieldPath, isNew, opts = {}) {
+  const {
+    onArray = defaultOnArray,
+    onObject = defaultOnObject
+  } = opts
```

Comment:
You could re-write this directly in the function signature like so:

```diff
-function targetFor (source, key, fieldPath, isNew, opts = {}) {
-  const {
-    onArray = defaultOnArray,
-    onObject = defaultOnObject
-  } = opts
-
+function targetFor (source, key, fieldPath, isNew, { onArray = defaultOnArray, onObject = defaultOnObject } = {}) {
```

Reply:
That's actually what I had originally, but it felt a bit absurdly long. 😅

Reply:
True that... was just thinking it was more "correct". Not sure if V8 cares though

Reply:
I think it would also improve auto-complete with VS Code and the like

Reply:
Ah, yeah. I suppose the auto-complete makes it worth it. 🤔